usdQt/opinionStackModel.py

```diff
@@ -0,0 +1,320 @@
+        super(OpinionStackModel, self).__init__(parent)
+        self.__handler = handler
+        self.ResetPrim(prim)
+
+    def __OnObjectsChanged(self, notice, sender):
+        resyncedPaths = notice.GetResyncedPaths()
+        changedInfoOnlyPaths = notice.GetChangedInfoOnlyPaths()
+
+        if not self.__prim and self.__valid:
+            # some change has caused the prim to expire
+            self.ResetPrim(None)
+            return
+
+        primPath = self.__prim.GetPath()
+        if primPath in resyncedPaths:
+            self.ResetPrim(self.__prim)
+        elif primPath in changedInfoOnlyPaths:
+            self.ResetPrim(self.__prim)
+
+    def _GetPrimTree(self, prim):
+        primStack = prim.GetPrimStack()
+        primDefinition = prim.GetPrimDefinition()
+        if primDefinition:
+            primStack.append(primDefinition)
+        primTree = []
+        for prim in primStack:
+            layer = prim.layer
+            if len(primTree) == 0 or layer != primTree[-1].layer:
+                primTree.append(_LayerItem(layer, len(primTree)))
+            primTree[-1].children.append(
+                _PrimItem(prim, primTree[-1]))
+            if not primTree[-1].strongestPrim:
+                if self.__handler.IsSpecified(prim):
+                    primTree[-1].strongestPrim = len(primTree[-1].children) - 1
+        return primTree
+
+    def ResetPrim(self, prim):
+        self.beginResetModel()
+        if prim is None:
+            self.__valid = False
+            self.__prim = Usd.Prim()
+        else:
+            self.__valid = True
+            self.__prim = prim
+            self.__listener = Tf.Notice.Register(
+                Usd.Notice.ObjectsChanged, self.__OnObjectsChanged, self.__prim.GetStage())
+
+        if self.__prim:
+            self.__primTree = self._GetPrimTree(prim)
+        else:
+            self.__primTree = []
+        self.endResetModel()
+
+    def parent(self, index):
+        if not index.isValid():
+            return QtCore.QModelIndex()
+        internalPointer = index.internalPointer()
+        if type(internalPointer) == _LayerItem:
+            return QtCore.QModelIndex()
+        else:
+            return self.createIndex(internalPointer.parent.row, 0, internalPointer.parent)
+
+    def headerData(self, section, orientation, role=QtCore.Qt.DisplayRole):
+        if orientation == QtCore.Qt.Horizontal:
+            if role == QtCore.Qt.DisplayRole:
+                if section == OpinionStackModel.SourceColumn:
+                    return "Source"
+                elif section == OpinionStackModel.OpinionColumn:
+                    return "Opinion"
+                else:
+                    raise Exception("invalid.")
+
+    def rowCount(self, parent=QtCore.QModelIndex()):
+        if not parent.isValid():
+            return len(self.__primTree)
+        internalPointer = parent.internalPointer()
+        if type(internalPointer) == _LayerItem:
```

Comment:
Use `is` when comparing types.  Also, using `isinstance` here is a little more flexible in case someone wants to author custom sub-classes of these item types.